CesiumIonClient/test/TestFillWithRandomBytes.cpp

```diff
@@ -0,0 +1,32 @@
+#include "../src/fillWithRandomBytes.h"
+
+#include <catch2/catch.hpp>
+
+using namespace CesiumIonClient;
+
+TEST_CASE("fillWithRandomBytes") {
+  std::vector<size_t> sizes = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+  for (size_t i = 0; i < sizes.size(); ++i) {
+    size_t size = sizes[i];
+
+    // Allocate an extra byte to make sure we don't overflow the buffer
+    std::vector<uint8_t> buffer(size + 1);
+    gsl::span<uint8_t> bufferSpan(buffer.data(), size);
+
+    fillWithRandomBytes(bufferSpan);
+
+    // Make sure the rest are non-zeros
+    for (size_t j = 0; j < size; ++j) {
+      // In the unlikely event the value is zero, generate some new random
+      // values. Repeat this up to 10 times. The chances that the random value
+      // at a particular position is zero ten times in a row is vanishingly
+      // small.
+      for (int k = 0; buffer[j] == 0 && k < 10; ++k)
+        fillWithRandomBytes(bufferSpan);
+      CHECK(buffer[j] != 0);
```

Comment:
Is the intent of this loop to make sure that we are able to generate a result with no zeros in any of the output, given at least 10 tries?
If so, line 25 might be resetting previous bytes in the buffer to zero that we will never check again.

Reply:
This is just a cheesy test. Testing for "actually random" isn't realistic. But I wanted the test to confirm that it doesn't accidentaly, for example, consistently leave the last byte zero due to an off-by-one error. So I check to make sure each byte isn't zero. But a random number generator can sometimes produce zeros, right? So if that happens, it just regenerates and expects that it won't be zero again. Unless there's an off-by-one error, and then that last byte might be zero all ten times and we'd have a bug, caught by the test. It doesn't matter if previous bytes end up being zero in the regeneration. The 10 is arbitrary. It's high enough that it's extraordinarily unlikely that we'll randomly get a zero in a particular position every time (1 in about 1e24 chance if my math is right). But low enough that we won't spin for ages if there's an actual bug.

Reply:
That's more clear thanks. 

I could see a future test that extends this to make sure that all possible values for a byte are represented in some way. Ex. I want to make sure that the value '5' appears at some point, not never showing up, but also in a random way.